gpkit/tools.py

```diff
@@ -28,6 +28,94 @@ def te_exp_minus1(posy, nterm):
         res += posy**i / factorial_denom
     return res
 
+def te_secant(x, nterm):
+    """Taylor expansion of secant(x).
+
+    Arguments
+    ---------
+    x : gpkit.monomial
+      Variable or expression argument
+    nterm : int
+        Number of terms in resulting Taylor expansion
+
+    Returns
+    -------
+    gpkit.Posynomial
+        Taylor expansion of secant(x), carried to nterm terms
+    """
+    if nterm < 1:
+        raise ValueError("Unexpected number of terms, nterm=%s" % nterm)
+
```

Comment:
Convert x to `radians` for automatic conversion in case x is in degrees or revolutions.